RFS/src/test/java/com/rfs/framework/ClusterOperations.java

```diff
@@ -145,4 +145,71 @@ public void createES6LegacyTemplate(final String templateName, final String patt
             );
         }
     }
+
+    /**
+     * Creates an ES7 legacy template, intended for use on only ES 7.8+ clusters
+     */
+    @SneakyThrows
+    public void createES7Templates(
+        final String componentTemplateName,
+        final String indexTemplateName,
+        final String fieldName,
+        final String indexPattern
+    ) throws IOException {
+        final var componentTemplateJson = "{"
+            + "\"template\": {"
+            + "    \"settings\": {"
+            + "        \"number_of_shards\": 1,"
+            + "        \"number_of_replicas\": 1"
+            + "    },"
+            + "    \"mappings\": {"
+            + "        \"properties\": {"
+            + "            \""
+            + fieldName
+            + "\": {"
+            + "                \"type\": \"text\""
+            + "            }"
+            + "        }"
+            + "    }"
+            + "},"
+            + "\"version\": 1"
+            + "}";
+
+        final var compTempUrl = clusterUrl + "/_component_template/" + componentTemplateName;
+        final var createCompTempRequest = new HttpPut(compTempUrl);
+        createCompTempRequest.setEntity(new StringEntity(componentTemplateJson));
+        createCompTempRequest.setHeader("Content-Type", "application/json");
+
+        try (var response = httpClient.execute(createCompTempRequest)) {
+            assertThat(
+                EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8),
+                response.getCode(),
+                equalTo(200)
+            );
+        }
+
+        final var indexTemplateJson = "{"
+            + "\"index_patterns\": [\""
+            + indexPattern
+            + "\"],"
+            + "\"composed_of\": [\""
+            + componentTemplateName
+            + "\"],"
+            + "\"priority\": 1,"
+            + "\"version\": 1"
+            + "}";
```

Comment:
Nit: I think this might be cleaner with string.format, but this is fine. 
![image](https://github.com/user-attachments/assets/f8e34dff-2138-4767-83f8-0b283ac071fd)

Reply:
Definitely not my first choice, but it makes Spotless happy...

Reply:
> Nit: I think this might be cleaner with string.format, but this is fine. ![image](https://private-user-images.githubusercontent.com/2754967/349230345-f8e34dff-2138-4767-83f8-0b283ac071fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExNTg3NzgsIm5iZiI6MTcyMTE1ODQ3OCwicGF0aCI6Ii8yNzU0OTY3LzM0OTIzMDM0NS1mOGUzNGRmZi0yMTM4LTQ3NjctODNmOC0wYjI4M2FjMDcxZmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTZUMTkzNDM4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjIwNjRiMWVmNDQwYjIzYTdhZGU3YjM0OWQ2YzE2ZDE1NGQwZTJkN2M1YzI5OTFkZjk2YjIyYjNmMWQzNWJlZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.H81l-C7XyzULbtBBJhE-lcz0A9XDbtZMIxV8RJvwmqA)

That would also be cleaner to grok w/ a map and jackson to output it.  Can the values ever contain any characters that could cause json an issue?